trtools/annotaTR/annotaTR.py

```diff
@@ -0,0 +1,597 @@
+        return 1
+    if args.dosages is None and args.ref_panel is None:
+        common.WARNING("No operation specified")
+        return 1
+
+
+    ###### Load reference panel info (optional) #######
+    refpanel_metadata = None
+    refreader = None
+    if args.ref_panel is not None:
+        common.MSG("Loading reference panel", debug=True)
+        refreader = utils.LoadSingleReader(args.ref_panel, lazy=True, samples=[])
+        if refreader is None:
+          return 1
+        if args.vcftype != 'auto':
+            refpanel_vcftype = trh.VcfTypes[args.vcftype]
+        else:
+            refpanel_vcftype = trh.InferVCFType(refreader)
+        if refpanel_vcftype == trh.VcfTypes.popstr:
+            common.WARNING("Error: reference panel annotation not "
+                           "currently supported for popSTR")
+            return 1
+        try:
+            match_on = RefMatchTypes[args.match_refpanel_on]
+        except KeyError:
+            common.WARNING("Invalid argument to --match-refpanel-on")
+            return 1
+        refpanel_metadata, ref_variant_ct = LoadMetadataFromRefPanel(refreader, refpanel_vcftype, 
+            match_on=match_on, ignore_duplicates=args.ignore_duplicates)
+        if len(refpanel_metadata.keys()) == 0:
+            common.WARNING("Error: No TRs detected in reference panel. Was the right vcftype specified? Quitting")
+            return 1
+        common.MSG("Loaded " + str(ref_variant_ct) + " TR loci from ref panel",
+            debug=True)
+
+    ###### Load reader #######
+    reader = utils.LoadSingleReader(args.vcf)
+    if reader is None:
+        return 1
+    if args.ref_panel is not None:
+        vcftype = refpanel_vcftype # should be same as refpanel
+    elif args.vcftype != 'auto':
+        vcftype = trh.VcfTypes[args.vcftype]
+    else:
+        vcftype = trh.InferVCFType(reader)
+
+    ##### More checks on input #####
+    # This check waits until here sicne need to have loaded reader
+    # and need to wait til refpanel loading to confirm vcf type
+    if dosage_type in [trh.TRDosageTypes.beagleap, trh.TRDosageTypes.beagleap_norm] \
+        and not trh.IsBeagleVCF(reader):
+        common.WARNING("Error: can only compute beagleap dosages on Beagle VCFs")
+        return 1
+
+    ###### Set up writers #######
+    # Update reader header, even if not writing VCF output
+    # This is because we might add VCF fields for parsing
+    # with TRHarmonizer along the way
+    if not UpdateVCFHeader(reader, " ".join(sys.argv), vcftype,
+                        dosage_type=dosage_type, refreader=refreader):
+        common.WARNING("Error: problem initializing vcf header.")
+        return 1
+    if OutputFileTypes.vcf in outtypes:
+        vcf_writer = cyvcf2.Writer(args.out+".vcf", reader)
+
+    # variant_ct needed for pgen
+    # If using a ref panel, assume we have same number
+    # of TRs as the panel
+    # Otherwise, assume our file is all TRs and use record count
```

Comment:
Good point! A mix of TRs and SNPs in the input file with no refpanel specified will result in an error when we try to convert a SNP record to a TRRecord. This would have resulted in a cryptic error before but I now added a check for this and will output a more helpful message before failing now:

```
"Error converting {chrom}:{pos} to a TR record. "
                "If your file is a mix of SNPs/TRs (e.g. from Beagle) you "
                "must provide a reference panel.".format(chrom=record.CHROM, pos=record.POS)
```